First LD round of edits

jhdalek55 opened the pull request:
Made changes in
History
Metadata
News
Overview
People
Security
Timeline

jhdalek55 committed:
First LD round of edits

Made changes in
History
Metadata
News
Overview
People
Security
Timeline

vladimir-v-diaz commented on metadata.md:
```diff
@@ -3,15 +3,25 @@ layout: default
+# Roles and Metadata
 
-Metadata files provide information that clients can use to make update decisions. Different metadata files provide different information. The various metadata files are signed by different keys as are indicated by the root role. The concept of roles allows TUF to only trust information that a role is trusted to provide.
+TUF uses metadata to create verifiable records about the state
```
Can we begin the paragraph with a discussion on roles, so that it matches its appearance in the section header?   Or you can switch their order in the header.

jhdalek55 replied:
fixed.

vladimir-v-diaz commented on metadata.md:
```diff
@@ -3,15 +3,25 @@ layout: default
+it to make decisions on which files to update. Different metadata files provide
+different information, and will be signed by different roles.
 
-The signed metadata files always include the time they were created and their expiration dates. This ensures that outdated metadata will be detected and that clients can refuse to accept metadata older than that which they've already seen.
```
> The signed metadata files always include **the time they were created** and their expiration dates.

Although once true, the time that metadata is created is no longer included in metadata.  We should instead say that, "The signed metadata files always include their expiration dates.

We should verify that the creation timestamp is not mentioned elsewhere in this document.

jhdalek55 replied:
fixed.

jhdalek55 replied:
I'll check for other references to the creation timestamp as I go forward.

vladimir-v-diaz commented on metadata.md:
```diff
@@ -3,15 +3,25 @@ layout: default
-The signed metadata files always include the time they were created and their expiration dates. This ensures that outdated metadata will be detected and that clients can refuse to accept metadata older than that which they've already seen.
+Roles define the set of actions a party can perform.The concept of roles allows
+TUF to only trust information provided by the correctly
+delegated party. The root role indicates which roles can sign for which projects.
```
In TUF, we use "delegated" and "delegation" to refer to a _delegated role_.  A Targets role can transfer trust of packages to another role, and we sometimes refer to the delegatee as a "delegated role."  I wonder if the use of "delegated party" might cause confusion.

jhdalek55 replied:
Changed

vladimir-v-diaz commented on metadata.md:
```diff
@@ -3,15 +3,25 @@ layout: default
-There are four required top-level roles and one optional top-level role, each with their own metadata file.
+All TUF metadata uses a subset of the JSON object format. When calculating the
+digest of an object, we use the [Canonical JSON](http://wiki.laptop.org/go/Canonical_JSON) format. Implementation-level detail about the metadata can be found in the [spec](docs/tuf-spec.txt).
+
```
The link to the specification is broken.  We should use the following URL: https://github.com/theupdateframework/specification/blob/master/tuf-spec.md

jhdalek55 replied:
fixed

vladimir-v-diaz commented on metadata.md:
```diff
@@ -28,58 +38,87 @@ There may also be any number of delegated target roles.
 Signed by: Targets role.
 
-The targets.json metadata file lists hashes and sizes of target files. Target files are the actual files that clients are intending to download (for example, the software updates they are trying to obtain).
+Target files are the actual files that client wants to download, such as
```
> Target files are the actual files **that client** wants to download, ...

Is this missing a word, or should it be "clients"?

jhdalek55 replied:
Fixed it. Should have been "actual files a client wants to download..."

jhdalek55 replied:
I went the plural to avoid a he/she thing. It's grammatically correct now.

vladimir-v-diaz commented on metadata.md:
```diff
@@ -28,58 +38,87 @@ There may also be any number of delegated target roles.
+available from the repository. When delegated roles are specified, it is done
+so in a way similar to how the Root role specifies the top-level roles: by giving
+the trusted keys and signature threshold for each role. Additionally, one or more
+patterns are specified to indicate the target file paths for which clients
```
Instead of "patterns," should we say "glob patterns" (perhaps with a Wikipedia link), to make it clear what is meant?

jhdalek55 replied:
You're the expert here, so i'll take your word for it. I changed the phrase and added the Wikipedia link

jhdalek55 replied:
changed the phrase and added a wikipedia link

vladimir-v-diaz commented on metadata.md:
```diff
@@ -28,58 +38,87 @@ There may also be any number of delegated target roles.
 Signed by: Snapshot role.
 
-The snapshot.json metadata file lists hashes and sizes of all metadata files other than timestamp.json. This file ensures that clients will see a consistent view of the files on the repository. That is, metadata files (and thus target file) that existed on the repository at different times cannot be combined and presented to clients by an attacker.
+The snapshot.json metadata file lists hashes and sizes of all metadata files
```
Although once true, Snapshot now lists the version number of all metadata files.  See the Snapshot [example](https://raw.githubusercontent.com/theupdateframework/tuf/develop/tests/repository_data/repository/metadata/snapshot.json).

jhdalek55 replied:
fixed

vladimir-v-diaz commented on metadata.md:
```diff
@@ -28,58 +38,87 @@ There may also be any number of delegated target roles.
-
-* The timestamp.json file is downloaded very frequently and so should be kept as small as possible, especially considering that the snapshot.json file grows in size in proportion to the number of delegated target roles.
-* As the Timestamp role's key is an online key and thus at high risk, separate keys should be used for signing the snapshot.json metadata file so that the Snapshot role's keys can be kept offline and thus more secure.
+The timestamp.json metadata file lists the hashes and size of the snapshot.json file.
```
The Timestamp file now also includes the version number of the Snapshot file.   See the Timestamp [example](https://raw.githubusercontent.com/theupdateframework/tuf/develop/tests/repository_data/repository/metadata/timestamp.json).

vladimir-v-diaz reviewed (commented):
Here is my review of the metadata.md edits.

vladimir-v-diaz commented on history.md:
```diff
@@ -3,33 +3,29 @@ layout: default
 css_id: history_timeline
```
We should probably edit the css_id.  `history.md` once included the timeline.

jhdalek55 replied:
These comments so far look great. I was going to go forward and do a little more work ---there more to be added to both news and press. Should I modify what's here first? And do I keep everything on this pull request as I go?

Sorry if these are dumb questions. I generally do a pull request at the end of each editing session, so I'm never sure about re-using the same PR

vladimir-v-diaz replied:
Yes, please keep everything on this pull request as you go.  In other words, continue pushing edits to this pull request.  This approach is easier for you and for reviewers.

Once you make edits that address one of my comments, my comment is "hidden."  I will also see your changes and can continue providing feedback.

vladimir-v-diaz replied:
In the future, you can try focusing each pull request to a particular file to minimize its size.

vladimir-v-diaz commented on metadata.md:
```diff
@@ -28,58 +38,87 @@ There may also be any number of delegated target roles.
+proportionally with the number of delegated target roles.
+* As the Timestamp role's key is an online key and thus at high risk, separate
+keys should be used for signing the snapshot.json metadata file so that the
+Snapshot role's keys can be kept offline, and thus more secure.
```
Adopters often wonder why we don't combine the Timestamp and Snapshot files.  Although we seem to cover it here, we should consider expanding the rationale since users still bring it up.

There is a discussion on our mailing list that covers it.
https://groups.google.com/forum/#!msg/theupdateframework/BoXjULlRNTw/gFtrmVPMAgAJ

jhdalek55 replied:
Maybe this is a change you should do, so we make sure its correct content-wise?

vladimir-v-diaz replied:
@JustinCappos Do you think the Timestamp section of `metadata.md` adequately explains why we separate Timestamp and Snapshot?

It seems to, although I suggest that we explicitly use "combine" when talking about the reasons for why these two roles are not one role.  It instead says, "There are two primary reasons why the timestamp.json file doesn't contain all of the information that the snapshot.json file does," which I think can be missed by some readers.

JustinCappos replied:
No, I don't like this as much.  Timestamp may be given to mirrors, etc. also.  I do think the size difference matters and is important, but snapshot often does need to be online.  

My feedback is mostly that I feel the emphasis is wrong, not so much the content.

vladimir-v-diaz replied:
Okay, I revised the text here: https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4/commits/2aaddacac1d6944c99e54deff639544efbcad0d4

@jhdalek55 Feel free to revise it, if needed.

jhdalek55 commented:
So, I'm confused. So, going forward should I do individual pull requests 
for each section or keep everything on this one? 


Lois

On Thu, Mar 01, 2018 at 11:20 AM, Vladimir Diaz wrote:

@vladimir-v-diaz commented on this pull request.




In history.md 
<https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#discussion_r171611829> 
:
> @@ -3,33 +3,29 @@ layout: default
  css_id: history_timeline


In the future, you can try focusing each pull request to a particular 
file to minimize its size.


—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub 
<https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#discussion_r171611829> 
, or mute the thread 
<https://github.com/notifications/unsubscribe-auth/AN5qX0wpk4N_AhHVAkrL7ZH-rVgeoltHks5taB-5gaJpZM4SXbuQ> 
.

vladimir-v-diaz commented:
Continue working on this pull request so that we finish it.  If you plan to make more changes to `news.md`, do them on this pull request (since it already includes edits to `news.md.`)

I was trying to say that when you do open a pull request in the future, try to keep it focused on one goal (for example, to a particular file).

jhdalek55 commented:
Ok. Understood.

On Thu, Mar 1, 2018 at 11:33 AM, Vladimir Diaz <notifications@github.com>
wrote:

> Continue working on this pull request so that we finish it. If you plan to
> make more changes to news.md, do them on this pull request (since it
> already includes edits to news.md.)
>
> I was trying to say that when you do open a pull request in the future,
> try to keep it focused on one goal (for example, to a particular file).
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#issuecomment-369650050>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AN5qX2HALVVrdUo4ctZ6816r7TLlswkCks5taCL2gaJpZM4SXbuQ>
> .
>

vladimir-v-diaz commented on timeline.md:
```diff
@@ -9,26 +9,58 @@ css_id: timeline
 
-**2015**: Docker launches Notary. It also launches Docker Content Trust, which
-uses Notary to sign and verify container images.
+**2014**: PEP 480, a maximum security version of PEP 458, is published
```
The entry doesn't end with a period, like the others.

jhdalek55 replied:
fixed

vladimir-v-diaz commented on timeline.md:
```diff
@@ -9,26 +9,58 @@ css_id: timeline
+of trust.
 
 **2016**: A consortium including NYU Tandon (Cappos, Kuppusamy, Diaz, Awwad),
 theUniversity of Michigan Transport Research Institute (UMTRI), and the
```
A space is needed in "theUniversity"

jhdalek55 replied:
Got it. Thanks.

vladimir-v-diaz commented on timeline.md:
```diff
@@ -9,26 +9,58 @@ css_id: timeline
+PyPI" details the integration of TUF into PyPI, the Python
+package manager.
+
 **2014**: Flynn.io independently implements TUF in the Go programming language.
```
We should consider using just "Flynn."  Flynn.io is the URL for the project.

Flynn is also the first organization to implement TUF (other than the reference implementation).  I'm not sure if that's captured in the current sentence.

jhdalek55 replied:
A very good point...one I wasn't aware of or it would have certainly received more emphasis.

vladimir-v-diaz commented on timeline.md:
```diff
@@ -9,26 +9,58 @@ css_id: timeline
 **2017**: *Mercury: Bandwidth-Effective Prevention of Rollback Attacks Against
-Community Repositories* accepted to USENIX ATC 2017.
+Community Repositories* accepted to USENIX ATC 2017. Mercury is another TUF
+adaptation developed to protect rollback attacks on community repositories at
```
Suggestion: "protect **against** rollback attacks ..."

jhdalek55 replied:
Oops...most have been getting late in the day. Good catch. fixed.

vladimir-v-diaz commented on timeline.md:
```diff
@@ -9,26 +9,58 @@ css_id: timeline
+a reasonable bandwidth cost.
+
+**2017**: Uptane is named one of the year's most important innovations in
+security by *Popular Science*
```
This entry also doesn't end with a period like the others.

jhdalek55 replied:
Just a general comment---I think we need to distinguish between the "timeline" and the "news" section. I'm thinking "timeline" can contain the major steps forward for TUF, while the "news" listings should be more current interest, but not necessarily important milestones. Towards that end, I'm thinking that the "news" section should not go back very far. FYI, do we still want to have events from 2016 on that list? If they are important enough to still be up, shouldn't they be "Timeline" events?

vladimir-v-diaz reviewed (commented):
Here is my review of `timeline.md`.

jhdalek55 committed:
Last corrections on this commit

Primarily these are corrections in response to Vlad’s review.

I will open a new PR for the remaining sections

vladimir-v-diaz commented:
@jhdalek55 
I still need to review the edits to `security.md` and `overview.md` (which I'm doing now).  I will open a separate pull request to explain why we don't combine Timestamp and Snapshot.

vladimir-v-diaz commented on overview.md:
```diff
@@ -4,81 +4,86 @@ css_id: overview
 ---
 
 ## Overview
+A quick look at what TUF does and how it works
```
This sentence doesn't end with a period.

jhdalek55 replied:
True. I was seeing this as a subhead, which is why I didn't add the period. Its fixed now

jhdalek55 commented:
That's fine.


Lois

On Thu, Mar 01, 2018 at 01:12 PM, Vladimir Diaz wrote:

@jhdalek55 <https://github.com/jhdalek55>

I still need to review the edits to security.md and overview.md (which 
I'm doing now).  I will open a separate pull request to explain why we 
don't combine Timestamp and Snapshot.


—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub 
<https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#issuecomment-369680524> 
, or mute the thread 
<https://github.com/notifications/unsubscribe-auth/AN5qX1RUxTDWrZHFU-UH-opOO_jcEE80ks5taDocgaJpZM4SXbuQ> 
.

vladimir-v-diaz commented on overview.md:
```diff
@@ -4,81 +4,86 @@ css_id: overview
+been introduced and used over the last decade or so to enhance the
+authenticity—and by extension, the security of update systems—most have drawbacks
+that have left update files and the repositories that host them.
+vulnerable to a number of attacks.
```
typo: `them. vulnerable ...`

jhdalek55 replied:
fixed.

vladimir-v-diaz commented on overview.md:
```diff
@@ -4,81 +4,86 @@ css_id: overview
+that can be used to secure new and existing software update systems.
 
-### Our approach ###
+* to provide the means to minimize the impact of key compromise
```
The entry doesn't end in with a period.

jhdalek55 replied:
Fixed.

vladimir-v-diaz commented on overview.md:
```diff
@@ -4,81 +4,86 @@ css_id: overview
-   PHP's Composer.
+TUF was launched almost a decade ago as a way to build system resilience against
+key compromises and other attacks that can spread malware or compromise a repository.
+The primary goals behind its design are:
```
Should we mention the specification?

jhdalek55 replied:
As a citation or in the text? We could do it either way. It probably should be mentioned here, just not sure how

vladimir-v-diaz replied:
In the text.  However, since these are design goals I'm not sure now if it makes sense to tie in the specification.

Perhaps we can toss it in the list for framework()? 

* to provide a framework (a set of libraries, file formats, **specification**, and utilities) that can be used to secure new and existing software update systems.

vladimir-v-diaz commented on overview.md:
```diff
@@ -4,81 +4,86 @@ css_id: overview
+* to be easy to integrate with existing software update systems.
 
-### Overview ###
+### Software Updates 101###
```
I think you need a space between the text and `###`, otherwise it shows up as:

```
Software Updates 101###
```

jhdalek55 replied:
Yup. Fixed.

vladimir-v-diaz commented on overview.md:
correction: "The Security page offers ..."

jhdalek55 replied:
Check. Fixed.

vladimir-v-diaz reviewed (commented):
Here's my review of `overview.md.`

vladimir-v-diaz commented on security.md:
typo: "An attacker who can compromises ..."

jhdalek55 replied:
fixed

jhdalek55 replied:
fixed

vladimir-v-diaz commented on security.md:
typo: "keysigning"

jhdalek55 replied:
fixed

vladimir-v-diaz commented on security.md:
suggestion: "... see the Metadata page" or "see metadata."

jhdalek55 replied:
fixed

vladimir-v-diaz commented on security.md:
> Trusting downloaded files really means assuming the files were provided by party without malicious designs.

Is this missing a word?

jhdalek55 replied:
should be "a party without..." Fixed.

vladimir-v-diaz commented on security.md:
A period is missing.

jhdalek55 replied:
Right. Fixed it.

vladimir-v-diaz reviewed (commented):
Okay, here are my remaining comments for this pull request.

vladimir-v-diaz committed:
Revise metadata.md's section on timestamp.json

Explicitly use "There are a few reasons why the timestamp.json and snapshot.json files are not combined:"

jhdalek55 committed:
Edits on press.md and reporting.md

jhdalek55 committed:
Merge pull request #1 from jhdalek55/second-inital-batch

Edits on press.md and reporting.md

vladimir-v-diaz commented on press.md:
```diff
@@ -7,27 +7,47 @@ css_id: press
 
 **[Lois, this page is not up to date.  Can you please add missing entries?]**
```
Let's not forget to remove this note once we feel the page is up to date.

jhdalek55 commented:
That makes sense. I'll fix it tomorrow.

Lois

On Thu, Mar 1, 2018 at 5:01 PM, Vladimir Diaz <notifications@github.com>
wrote:

> *@vladimir-v-diaz* commented on this pull request.
> ------------------------------
>
> In overview.md
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#discussion_r171708016>
> :
>
> >
> -* **Library package managers** such as those offered by many
> -   programming languages for installing additional libraries. These are systems
> -   such as Python's pip/easy_install + PyPI, Perl's CPAN, Ruby's RubyGems, and
> -   PHP's Composer.
> +TUF was launched almost a decade ago as a way to build system resilience against
> +key compromises and other attacks that can spread malware or compromise a repository.
> +The primary goals behind its design are:
>
> In the text. However, since these are design goals I'm not sure now if it
> makes sense to tie in the specification.
>
> Perhaps we can toss it in the list for framework()?
>
>    - to provide a framework (a set of libraries, file formats,
>    *specification*, and utilities) that can be used to secure new and
>    existing software update systems.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#discussion_r171708016>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AN5qX1um71L0l7x76iOUwkh0DZy2L-c_ks5taG_MgaJpZM4SXbuQ>
> .
>

vladimir-v-diaz commented:
> That makes sense. I'll fix it tomorrow.

Does this pull request still need work?  Do you think what we have so far (after merging this PR) is good enough to push to the live website?  We can still continue making improvements after going live.

@JustinCappos will also let us know if the current state of the redesign is good enough for the live website.

jhdalek55 commented:
I  looked at it again last week and I think the pull request is good as is.
It may still need some refinements, but the type of fine-tuning I'm talking
about can't be done till the changes are incorporated and there's a clean
draft to review. I'll defer to Justin as to whether he thinks its ready to
go live. I know I'd probably like to give the clean copy one more look
myself before going live.

Lois

On Tue, Mar 20, 2018 at 11:22 AM, Vladimir Diaz <notifications@github.com>
wrote:

> That makes sense. I'll fix it tomorrow.
>
> Does this pull request still need work? Do you think what we have so far
> (after merging this PR) is good enough to push to the live website? We can
> still continue making improvements after going live.
>
> @JustinCappos <https://github.com/justincappos> will also let us know if
> the current state of the redesign is good enough for the live website.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#issuecomment-374638134>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AN5qX93Exb25lSa-0DwJLpRo8AFDF5Nrks5tgR6vgaJpZM4SXbuQ>
> .
>

vladimir-v-diaz commented:
Thanks for the quick reply!  Okay, I'll go ahead and merge this pull request so that Justin can take a look and you can have a clean draft to review.  You are welcome to submit another pull request with edits, if needed.

jhdalek55 commented:
Will do. I don't anticipate a lot of edits. Mainly its polishing and
proofreading I want to do, just to make sure in moving things around we
didn't leave anything incomplete or messy.

I'm working on something else right ow. Will try to take a look at this
next week.

Lois

On Tue, Mar 20, 2018 at 11:33 AM, Vladimir Diaz <notifications@github.com>
wrote:

> Merged #4
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4>.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#event-1531230553>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AN5qX9XpbjJjzzuLO4rlYiyFpKmXcW75ks5tgSFTgaJpZM4SXbuQ>
> .
>

JustinCappos commented:
Okay, I've taken a look.  It looks nice!  You can go ahead and move this to
production.  (We can refine further from there.)


On Tue, Mar 20, 2018 at 11:36 AM, Lois Anne DeLong <notifications@github.com
> wrote:

> Will do. I don't anticipate a lot of edits. Mainly its polishing and
> proofreading I want to do, just to make sure in moving things around we
> didn't leave anything incomplete or messy.
>
> I'm working on something else right ow. Will try to take a look at this
> next week.
>
> Lois
>
> On Tue, Mar 20, 2018 at 11:33 AM, Vladimir Diaz <notifications@github.com>
> wrote:
>
> > Merged #4
> > <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4>.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.
> io/pull/4#event-1531230553>,
> > or mute the thread
> > <https://github.com/notifications/unsubscribe-auth/
> AN5qX9XpbjJjzzuLO4rlYiyFpKmXcW75ks5tgSFTgaJpZM4SXbuQ>
>
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/vladimir-v-diaz/vladimir-v-diaz.github.io/pull/4#issuecomment-374643418>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AA0XDxvG9aN0zxSNrNB1u1ANFNyodwuJks5tgSHwgaJpZM4SXbuQ>
> .
>
